developer-docs-site/docs/guides/sign-a-transaction.md

```diff
@@ -0,0 +1,300 @@
+
+* Prepare the unsigned transaction, known as a `RawTransaction`.
+* Generate the appropriate salt for signing the transaction.
+* Sign this salt and the transaction to produce an `Authenticator`, and
+* Derive a complete and signed transaction, consisting of an  `Authenticator` and a `RawTransaction`.
+
+See the below diagram showing a high-level flow.
+
+![creating-signed-transaction.svg](/img/docs/creating-signed-transaction.svg)
+
+Unsigned transactions are known as `RawTransaction`s. They contain all the information about how to execute an operation on an account within Aptos. But they lack the appropriate authorization with a signature or `Authenticator`.
+
+In Aptos Blockchain, all the data is encoded as [BCS][bcs] (Binary Canonical Serialization). Aptos supports many different approaches to signing but defaults to a single signer using [Ed25519][Ed25519]. The `Authenticator` produced during the signing gives authorization to the Aptos Blockchain to execute the transaction on behalf of the account owner.
+
+## Key concepts
+
+### Raw transaction
+
+Raw transactions consist of the following fields:
+
+* **sender** Account address of the sender.
+* **sequence_number** (uint64): Sequence number of this transaction. This must match the sequence number stored in the sender's account at the time the transaction executes.
+* **payload**: Instructions for the Aptos Blockchain, including publishing a module, execute a script function or execute a script payload.
+* **max_gas_amount** (uint64): Maximum total gas to spend for this transaction. The account must have more than this gas or the transaction will be discarded during validation.
+* **gas_unit_price** (uint64): Price to be paid per gas unit. During execution the `total_gas_amount`, calculated as: `total_gas_amount = txn_base_cost * gas_unit_price`, must not exceed `max_gas_amount` or the transaction will abort during the execution. `txn_base_cost` represents the total units of gas consumed when executing the transaction.
+* **expiration_timestamp_secs** (uint64): The blockchain timestamp at which the blockchain would discard this transaction.
+* **chain_id** (uint8): The chain ID of the blockchain that this transaction is intended to be run on.
+
+### BCS
+
+Binary Canonical Serialization (BCS) is a serialization format. See [BCS][bcs] for a description of the design goals of BCS.
+
+:::note
+
+BCS does not support floats, unicode chars and sets.
+
+:::
+
+BCS is not a self-describing format. As such, in order to deserialize a message, one must know the message type and layout ahead of time.
+
+An example of how BCS serializes a string.
+```typescript
+// A string is serialized as: byte length + byte representation of the string. The byte length is required for deserialization. Without it, no way the deserializer knows how many bytes to deserialize.
+const bytes: Unint8Array = bcs_serialize_string("aptos");
+assert(bytes == [5, 0x61, 0x70, 0x74, 0x6F, 0x73]);
+```
+
+### Signing message
```

Comment:
Maybe

Safe Message Signing

Reply:
"Signing message" is a term used in our REST api path and spec. I would rather leave as is.